### PR TITLE
[Bug](materialized-view) fix problems about create mv on ssb_flat q4.1 failed

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

fix problems about create mv on ssb q4.1 failed

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by yiguolei** [comment]
Maybe we do not need modify old planner, it is very dangerous to maintain the new materialized view in old planner. Just modify nerids planner. It is safe.

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/functions/function.h [hunk 1]
```diff
@@ -314,6 +315,17 @@ class IFunctionBuilder {
 
 using FunctionBuilderPtr = std::shared_ptr<IFunctionBuilder>;
 
+inline std::string get_types_string(const ColumnsWithTypeAndName& arguments) {
+    std::string types;
+    for (auto argument : arguments) {
```
Comment (HappenLee) [inline be/src/vec/functions/function.h:320]: const auto&

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Comment by qg-lin** [comment]
@BiteTheDDDDt Hi, is there any way to disable funtion return type check?

**Comment by BiteTheDDDDt** [comment]
> @BiteTheDDDDt Hi, is there any way to disable funtion return type check?

This check should not be disabled. Incorrect types may cause unexpected errors.

**Pull request merged**